Add deletion button for citations

gaurav opened the pull request:
This PR adds a delete button to citations, while also tweaking the display of citations. It also fixes a bug in PhylorefView, where we referred to `definitionSource` as `obo:IAO_0000119`, which was its pre-publication field name. Closes #167.

Citation filled in:
<img width="1145" alt="Screen Shot 2022-04-26 at 9 46 00 PM" src="https://user-images.githubusercontent.com/23979/165421791-3617fbcb-1c6a-4a75-a912-2f211a29d6f4.png">

Citation expanded:
<img width="1555" alt="Screen Shot 2022-04-26 at 9 46 25 PM" src="https://user-images.githubusercontent.com/23979/165421826-f388bc3c-295b-419b-a6d3-af295f183bcf.png">

Pressing the red 'X' at the far right displays a confirmation dialog:
<img width="383" alt="Screen Shot 2022-03-29 at 1 29 05 AM" src="https://user-images.githubusercontent.com/23979/160539821-7782a747-cea2-4bc1-9d00-b3db48bd067e.png">

If the user selects "OK", the citation is deleted:
<img width="1145" alt="Screen Shot 2022-04-26 at 9 47 43 PM" src="https://user-images.githubusercontent.com/23979/165421915-723a765b-40bf-4cf8-ab72-af098c218ec9.png">

The "Add citation" text is underlined if the user hovers over it:
<img width="1142" alt="Screen Shot 2022-04-26 at 9 48 28 PM" src="https://user-images.githubusercontent.com/23979/165422031-0d517cd5-18cb-4bba-82f7-1caf4374c5fd.png">

hlapp reviewed (commented):
So instead of &times; can't we use a trash bin icon (should be available within font-awesome, if we're using that?)? The cross is just often used for closing something, or making something  disappear from view. A trash bin would be unequivocal.

gaurav commented:
Updated with the trash icon from Bootstrap -- the full list of icons available to us are at https://bootstrap-vue.org/docs/icons#icons-1. I've also updated the screenshots with the new trash icon.

hlapp reviewed (approved):
Looks good!
